packages/actor-init-query/lib/HttpServiceSparqlEndpoint.ts

```diff
@@ -426,6 +432,10 @@ export class HttpServiceSparqlEndpoint {
     readOnly: boolean,
     queryId: number,
   ): Promise<void> {
+    if (this.includeVoID && request.url?.split('/').at(-1)?.startsWith('void')) {
+      return this.writeVoIDDescription(engine, stdout, stderr, request, response, mediaType, headOnly);
```

Comment:
The `writeServiceDescription` method also produces a set of quads as the output, so I was wondering if it would be feasible to merge the VoID description output into the service description? 🤔 

Quite a few endpoints, for example Wikidata and UniProt, that expose a VoID description, do so within the service description, so there is no need to follow additional links or try additional URIs. The [VoID specification](https://www.w3.org/TR/void/#sparql-sd) also has it as an example publishing format. It is also the cleanest option of the ones offered by the spec (section 6.2).

Reply:
For reference, you can get the Wikidata and UniProt SD + VoID with curl:
```bash
curl -o ./wikidata-sd.nt -H "accept: application/n-triples" https://query.wikidata.org/sparql
curl -o ./uniprot-sd.nt -H "accept: application/n-triples" https://sparql.uniprot.org/sparql
```
They take slightly different approaches, but if I remember correctly, they both have a similar link path from the service to the (named or union) default graph, and/or the default dataset, which has the VoID description associated with it.

Reply:
The Comunica code to extract VoID description data is [here](https://github.com/comunica/comunica/blob/e943589fbe08c99c270019d5b2772288d17fb48b/packages/actor-rdf-metadata-extract-void/lib/ActorRdfMetadataExtractVoid.ts#L52) in `ActorRdfMetadataExtractVoid`.